build-tools/packages/version-tools/src/versions.ts

```diff
@@ -19,21 +24,25 @@ export function getSimpleVersion(
 	fileVersion: string,
 	argBuildNum: string,
 	argRelease: boolean,
-	patch: boolean,
+	simplePatchVersioning: boolean,
 ) {
 	// Azure DevOp passes in the build number as $(buildNum).$(buildAttempt).
 	// Get the Build number and ignore the attempt number.
-	const buildId = patch ? parseInt(argBuildNum.split(".")[0], 10) : undefined;
+	const buildId = simplePatchVersioning ? parseInt(argBuildNum.split(".")[0], 10) : undefined;
 	let version = fileVersion;
 	if (isInternalVersionScheme(version, /* allowPrereleases */ true)) {
-		if (patch) {
+		if (simplePatchVersioning) {
 			throw new Error(
 				`Cannot use simple patch versioning with Fluid internal versions. Version: ${fileVersion}`,
 			);
 		}
 
 		if (!argRelease) {
-			version = changePreReleaseIdentifier(version, "dev");
+			const [, , prereleaseId] = fromInternalScheme(version);
+			version = changePreReleaseIdentifier(
+				version,
+				prereleaseId === ALLOWED_PRERELEASE_IDENTIFIERS[0] ? "dev" : `dev-${prereleaseId}`,
```

Comment:
>ALLOWED_PRERELEASE_IDENTIFIERS[0] [](http://example.com/codeflow?start=21&length=33)

This seems suspect.
Would be very easy to change array and have this change for good or bad.
Suggest breaking out [0] to be its own named constant (and hopefully clear intent and reasonable use).

Reply:
I wrote it this way so that it was easier to maintain if the prerelease IDs change (they have changed twice while this
PR has been open). I could change it now, but I'm inclined to leave that for a future change. I'm not 100% convinced we
will ship with "rc" as the final prerelease ID. :)

Reply:
Sorry commented on the wrong feedback - fixing.